@navikt/core/tokens/darkside/figma/plugin/AkselVariablesInterface.ts

```diff
@@ -0,0 +1,258 @@
+      /* Color values can only be defined by strings */
+      if (typeof token.value !== "string") {
+        throw new Error(`Token value is not a string: ${token}`);
+      }
+
+      let variable = this.Figma.getVariable(token.name, collection.id);
+
+      if (!variable) {
+        variable = this.Figma.createVariable(
+          token.name,
+          collection,
+          token.figmaType,
+        );
+      }
+
+      this.Figma.setVariableValue(
+        variable,
+        figma.util.rgba(token.value),
+        collection.defaultModeId,
+      );
+
+      this.Figma.setVariableMetadata(variable, {
+        codeSyntax: token.code,
+        description: token.comment ?? "",
+        hiddenFromPublishing: collection.hiddenFromPublishing,
+        scopes: token.scopes,
+      });
+    }
+
+    console.info("Updated collection: ", collection.name);
+  }
+
+  private updateScaleCollection(
+    globalScale:
+      | ScopedFigmaTokenConfig["radius"]
+      | ScopedFigmaTokenConfig["spacing"],
+  ): void {
+    let collection = this.Figma.getCollection(globalScale.name);
+
+    if (!collection) {
+      collection = this.Figma.createCollection(globalScale.name);
+    }
+
+    const sortedTokens = globalScale.tokens.sort((a, b) => {
+      if (typeof a.value === "number" && typeof b.value === "number") {
+        return a.value - b.value;
+      }
+      return 0;
+    });
+
+    for (const token of sortedTokens) {
+      let variable = this.Figma.getVariable(token.name, collection.id);
+
+      if (!variable) {
+        variable = this.Figma.createVariable(
+          token.name,
+          collection,
+          token.figmaType,
+        );
+      }
+
+      this.Figma.setVariableValue(
+        variable,
+        token.value,
+        collection.defaultModeId,
+      );
+
+      this.Figma.setVariableMetadata(variable, {
+        codeSyntax: token.code,
+        description: token.comment ?? "",
+        hiddenFromPublishing: collection.hiddenFromPublishing,
+        scopes: token.scopes,
+      });
+    }
+
+    console.info("Updated collection: ", collection.name);
+  }
+
+  private updateSemanticColorCollection(
+    colorsConfig: ScopedFigmaTokenConfig["colors"],
+  ): void {
+    const semanticCollectionName = colorsConfig.light.semantic.name;
+
+    let collection = this.Figma.getCollection(semanticCollectionName);
+
+    if (!collection) {
+      collection = this.Figma.createCollection(semanticCollectionName);
+    }
+
+    for (const modeName of Object.values(colorsConfig).map((x) => x.name)) {
+      const globalCollection = this.Figma.getCollection(
+        colorsConfig[modeName].global.name,
+      );
+
+      if (!globalCollection) {
+        throw new Error(
+          `Global collection not found for: ${colorsConfig[modeName].global.name}`,
+        );
+      }
+
+      let mode = this.Figma.getModeWithName(modeName, collection);
+
+      if (!mode) {
+        mode = this.Figma.createMode(modeName, collection);
+      }
```

Comment:
Would this work? 🤔
```suggestion
      const mode =
        this.Figma.getModeWithName(modeName, collection) ??
        this.Figma.createMode(modeName, collection);
```
If it works here, maybe also for get/createCollection.

You could also consider just merging `getModeWithName` and `createMode` into `getOrCreateMode` since they are always used together.

Reply:
That works, updated variables and collections to also do this 🥇 

I feel this is a good middleground for not being to verbose, while still being easy to read and understand without hiding any "magic" behind some composite functions.
```tsx
const variable =
          this.Figma.getVariable(token.name, collection.id) ??
          this.Figma.createVariable(token.name, collection, token.figmaType);
```